File: src/Controlador/ControladorPeriodo.java
package Controlador;

import java.util.Scanner;

import Leitor.LeitorPeriodo;
import Modelo.Periodo;

public class ControladorPeriodo implements IControlador{
	
    public void ler(Scanner scan) throws Exception {
		LeitorPeriodo leitor = LeitorPeriodo.obterInstancia();
		while(scan.hasNext()){
			Periodo periodo = leitor.ler(scan);
			leitor.anexaHash(periodo);
		}
	}

	public Periodo busca(String codigo){
		LeitorPeriodo leitor = LeitorPeriodo.obterInstancia();
		return leitor.busca(codigo);
	}
}

File: cpp_version/Controlador/ControladorPeriodo.cpp
#include "ControladorPeriodo.h"

Periodo* ControladorPeriodo :: busca(string codigo){
    LeitorPeriodo *leitor= LeitorPeriodo :: obterInstancia();
    return leitor->busca(codigo);
}

vector<Periodo*> ControladorPeriodo :: lista(){
    LeitorPeriodo *leitor= LeitorPeriodo :: obterInstancia();
    return leitor->obterPeriodos();
}

void ControladorPeriodo ::ler(ifstream* scan){
    
    LeitorPeriodo* leitor = LeitorPeriodo::obterInstancia();

    string linha;
    getline(*scan, linha);
    while (getline(*scan, linha)) {
        Tokenizer tok(linha, ';');
        vector<string> vec = tok.remaining();
        for (int i = 0; i < vec.size(); i++){
            vec[i] = trim(vec[i]);
        }
        Periodo* periodo = leitor->ler(vec);
        leitor->anexaHash(periodo);
    }
}
